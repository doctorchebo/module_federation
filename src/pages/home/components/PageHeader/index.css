.ui.container.header-container {
	height: 80vh;
}

.ui.inverted.header.main-title {
	font-size: 6em;
	font-weight: normal;
	margin-bottom: 0;
	margin-top: 1.7em;
	color: #fff;
	overflow-y: scroll;
	animation: burbujas 5s linear infinite;
	height: auto;
}

.ui.header.sub-title {
	font-size: 2.3em;
	font-weight: normal;
	margin-top: 0.2em;
	color: rgb(66, 55, 55);
}

@keyframes burbujas {
	0% {
		bottom: 0;
		opacity: 0;
	}
	50% {
		opacity: 2;
	}
	100% {
		bottom: 100vh;
		opacity: 0;
	}
}
