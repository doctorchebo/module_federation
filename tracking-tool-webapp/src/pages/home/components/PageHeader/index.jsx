import React from 'react';
import PropTypes from 'prop-types';
import './index.css';

import { Container, Header } from 'semantic-ui-react';

/**
 *
 * @param {object} props properties
 * @returns {React.Component} Page header
 */
export default function PageHeader(props) {
	const { appName, subtitle } = props.constants;
	return (
		<Container className='header-container'>
			<Header inverted as='h2' content={appName} className='main-title' />
			<Header as='h3' className='sub-title' content={subtitle} />
		</Container>
	);
}

PageHeader.propTypes = {
	constants: PropTypes.object,
	headerAppName: PropTypes.string,
	headerSubTitle: PropTypes.string,
};
