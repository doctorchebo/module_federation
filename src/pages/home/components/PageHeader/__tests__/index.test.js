import React from 'react';
import PageHeader from '../.';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

const headerConstants = {
	appName: 'Applicants Tracker',
	subtitle: 'An app for applicants in the Software Industry.',
	getStarted: 'Get Started',
};

describe('pages/home/components/PageHeader', () => {
	describe('Html structure', () => {
		it('should render with default props', () => {
			const { container } = render(<PageHeader constants={headerConstants} />, {
				wrapper: MemoryRouter,
			});
			expect(container.firstChild).toBeInstanceOf(HTMLDivElement);
		});

		it('should match snapshot', () => {
			const { container } = render(<PageHeader constants={headerConstants} />, {
				wrapper: MemoryRouter,
			});
			expect(container.firstChild).toMatchSnapshot();
		});

		it('header should be instance of header', () => {
			const { container } = render(<PageHeader constants={headerConstants} />, {
				wrapper: MemoryRouter,
			});
			const header = container.querySelector('.ui.inverted.header.main-title');
			expect(header).toBeInstanceOf(HTMLHeadingElement);
		});

		it('header content should be what is expected', () => {
			const { container } = render(<PageHeader constants={headerConstants} />, {
				wrapper: MemoryRouter,
			});
			const header = container.querySelector('.ui.inverted.header.main-title');
			expect(header.innerHTML).toBe(headerConstants.appName);
		});
		it('h2 should be instance of header', () => {
			const { container } = render(<PageHeader constants={headerConstants} />, {
				wrapper: MemoryRouter,
			});
			const header = container.querySelector('.ui.header.sub-title');
			expect(header).toBeInstanceOf(HTMLHeadingElement);
		});

		it('h2 content should be what is expected', () => {
			const { container } = render(<PageHeader constants={headerConstants} />, {
				wrapper: MemoryRouter,
			});
			const header = container.querySelector('.ui.header.sub-title');
			expect(header.innerHTML).toBe(headerConstants.subtitle);
		});
	});
});
