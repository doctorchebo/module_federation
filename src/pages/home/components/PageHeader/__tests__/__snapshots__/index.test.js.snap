// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/home/components/PageHeader Html structure should match snapshot 1`] = `
<div
  class="ui container header-container"
>
  <h2
    class="ui inverted header main-title"
  >
    Applicants Tracker
  </h2>
  <h3
    class="ui header sub-title"
  >
    An app for applicants in the Software Industry.
  </h3>
</div>
`;
